paramiko/proxy.py

```diff
@@ -105,7 +105,8 @@ def recv(self, size):
             raise ProxyCommandFailure(' '.join(self.cmd), e.strerror)
 
     def close(self):
-        os.kill(self.process.pid, signal.SIGTERM)
+        self.process.kill()
+        self.process.poll()
```

Comment:
This should be "self.process.wait()".  In my tests, poll() doesn't' remove the zombie (which makes sense if you read the man page (2) for wait()).